### DISCUSSION TITLE
Model is speaking differently in the web ui compared to the API

### DISCUSSION THREAD

**Discussion by gdet** [oobabooga/text-generation-webui#6450] [General]
Hello everyone,
I have loaded a local model and it is working very well in the chat tab but when I am using the API it is not speaking so good.
I tried to pass as many arguments that I found but still there are issues.

     payload = {
        "messages": history,  # The user's input with the history
        "mode": "chat-instruct", 
        "character": "Assistant",
        "greeting": "Hello. I would like to ask you some questions",
        "max_new_tokens": 512,  # Adjust as necessary     # Adjust as necessary
        "stop": ["\n"],         # Define the stop tokens as needed
        "do_sample": True,      # Set to False for deterministic outpu       
        "temperature": 0.85,
        'top_p': 1,
        "top_k":0,
        'typical_p': 1,
        'min_p':0.05,
        'repetition_penalty': 1.01,
        'encoder_repetition_penalty': 1,
        'presence_penalty':0,
        'frequency_penalty':0,
        'repetition_penalty_range':1024,
        'top_k': 50,
        'min_length': 0,
        'no_repeat_ngram_size': 0,
        'num_beams': 1,
        'penalty_alpha': 0,
        'length_penalty': 1,
        'early_stopping': False,
        'seed': -1,
        'add_bos_token': True,
        'truncation_length': 2048,
        'ban_eos_token': False,
        'attn_implementation':'eager',
        'torch_dtype':'bf16',
         "seed": 42
    }

I also added the same chat_instruct_command but it does not improve.Also greeting is not working. Sometimes it is ok and sometimes it is not ok at all.
What else can I do? Am I missing something?
In model tab there is eager_attention enabled and bf16 but I do not know how to pass them as parameters.
Thank you

**Comment by TiagoTiago** [comment]
Set the WebUI to verbose mode, and look at the parameters it outputs in the console, see if there's anything there different from what you're using in your API call.
